src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -66,56 +66,63 @@ def killOlderProcesses(self):
-                                                     print("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output")
-                                                     Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output", user='zookeeper')
-                                                     status, output = commands.getstatusoutput("tail -n1 /tmp/twill_" + service + "_instances_output")
-                                                     if status == 0:
-                                                         print(output)
-                                                         data = json.loads(output)
-                                                         container_id = data['data']['containerId']
-                                                         container_values = container_id.split('_')
-                                                         applicationId = "application_" + container_values[2] + "_" + container_values[3] 
-                                                         kinit_cmd = params.kinit_cmd_master
-                                                         kill_application_cmd = format("{kinit_cmd} yarn application -kill " + applicationId)
-                                                         print(kill_application_cmd)
-                                                         Execute(kill_application_cmd, user='cdap')
-                                                         print("applicationId : " + applicationId)
-                                                     Execute("rm -rf /tmp/twill_" + service + "_instances_output", user='zookeeper') 
-                                         Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper') 
-                    Execute("rm -rf /tmp/twill_master_services", user='zookeeper')                  
+        if status != 0:
+            print('fail in executing the command ls /election/master.services on zookeeper')
+            return
+        if (!(output.startswith('[') and output.endswith(']') and output == '[]')):
+            print('outt=put value is not equal to []')
+            return
+        # exectute ls /cdap/twill and save its output
+        Execute("/usr/hdp/3.1.0.0-78/zookeeper/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " ls /twill > /tmp/twill_master_services", user='zookeeper')
+        status, output = commands.getstatusoutput("tail -n1 /tmp/twill_master_services")
+        if status != 0:
+            print('fail in executing the command ls /twill on zookeeper')
+            return
+        if (!(output.startswith('[') and output.endswith(']') and output != '[]')):
+            print('output value is equal to []')
+            return
```

Comment:
We are executing a command, redirecting its output to a file, capturing the output by tailing file and then status of tail as well as output here. Instead of sending output to file, can we directly capture the output from 'Execute' call ? This will remove statements and shorten the code. 
Also, I see similar thing being done for multiple other commands below so looks like a lot of duplicate code. Can we create a separate function to remove duplicate code ?

Reply:
Created a common function to be called